### PR TITLE
Moving cluster to new framework part

### PR DESCRIPTION
**Pull request by tsorya**
Moving cluster to new framework part

In order to see diff please look on second commit

### CODE REVIEW

File: provider/cluster/cluster_resource.go [hunk 1]
```diff
@@ -0,0 +1,706 @@
...
+				Description: "AWS access key.",
+				Optional:    true,
+				Sensitive:   true,
+			},
+			"aws_subnet_ids": schema.ListAttribute{
+				Description: "AWS subnet IDs.",
+				ElementType: types.StringType,
+				Optional:    true,
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.RequiresReplace(),
```
Comment (bardielle) [inline provider/cluster/cluster_resource.go:144]: It wasn't set as RequiresReplace - Just to make sureyou meant for that

File: provider/cluster/cluster_resource.go [hunk 2]
```diff
@@ -0,0 +1,706 @@
...
+					boolplanmodifier.UseStateForUnknown(),
+					boolplanmodifier.RequiresReplace(),
+				},
+			},
+			"availability_zones": schema.ListAttribute{
+				Description: "Availability zones.",
+				ElementType: types.StringType,
+				Optional:    true,
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.UseStateForUnknown(),
```
Comment (bardielle) [inline provider/cluster/cluster_resource.go:161]: Did you meant to wrote it? it wasn't set in the past

File: provider/cluster/cluster_resource.go [hunk 3]
```diff
@@ -0,0 +1,706 @@
...
+					listplanmodifier.RequiresReplace(),
+				},
+			},
+			"machine_cidr": schema.StringAttribute{
+				Description: "Block of IP addresses for nodes.",
+				Optional:    true,
+				Computed:    true,
+			},
+			"proxy": schema.SingleNestedAttribute{
```
Comment (bardielle) [inline provider/cluster/cluster_resource.go:170]: Can we expose it? to use the same single Nested from the cluaterrosa classic?

File: provider/cluster/cluster_resource.go [hunk 4]
```diff
@@ -0,0 +1,706 @@
...
+				Description: "proxy",
+				Attributes: map[string]schema.Attribute{
+					"http_proxy": schema.StringAttribute{
+						Description: "HTTP proxy.",
+						Optional:    true,
```
Comment (bardielle) [inline provider/cluster/cluster_resource.go:175]: In the past it was required
```
					"http_proxy": {
						Description: "http proxy",
						Type:        types.StringType,
						Required:    true,
					},
```

File: provider/cluster/cluster_resource.go [hunk 5]
```diff
@@ -0,0 +1,706 @@
...
+					},
+					"https_proxy": schema.StringAttribute{
```
Comment (bardielle) [inline provider/cluster/cluster_resource.go:177]: In the past it was required

File: provider/cluster/cluster_resource.go [hunk 6]
```diff
@@ -0,0 +1,706 @@
...
+				"Can't create cluster with name '%s': %v",
+				state.Name.ValueString(), err,
+			),
+		)
+		return
+	}
+	object = add.Body()
+
+	// Wait till the cluster is ready unless explicitly disabled:
+	wait := state.Wait.IsUnknown() || state.Wait.IsNull() || state.Wait.ValueBool()
```
Comment (bardielle) [inline provider/cluster/cluster_resource.go:393]: HasValue

File: provider/provider.go [hunk 7]
```diff
@@ -213,6 +214,7 @@ func (p *Provider) Resources(ctx context.Context) []func() resource.Resource {
 		oidcconfig.New,
 		oidcconfiginput.New,
 		clusterrosaclassic.New,
+		cluster.New,
 		// TODO uncomment this after ClusterRosaClassic resource is fixed
 		// clusterwaiter.NewClusterRosaClassicResource,
```
Comment (bardielle) [inline provider/provider.go:219]: use the updated provider.go

File: provider/cluster/cluster_resource.go (hunk 1 above)
Comment (tsorya) [reply provider/cluster/cluster_resource.go:144]: Took from rosa classic and i assume it is right

File: provider/cluster/cluster_resource.go (hunk 2 above)
Comment (tsorya) [reply provider/cluster/cluster_resource.go:161]: Took from rosa classic and i assume it is right

**Comment by bardielle** [comment]
@tsorya I opened a new ticket - 

[OCM-3740](https://issues.redhat.com/browse/OCM-3740) | availability zones default in cluster_resource and cluster_rosa_classic

**Pull request merged**